File: 344-ReverseString/344-ReverseString.cpp
// Last updated: 16/10/2025, 04:27:21
class Solution {
public:
    
    
    void reverseString(vector<char>& s) {
        stack<int>st;
        for(int i=0; i<s.size(); i++){
            st.push(s[i]);
        }
        int count = 0;
        while(count<s.size()){
             s[count] = st.top();
             st.pop();
            count++;
        }
    }
};

File: 344-ReverseString/344-ReverseString.java
// Last updated: 16/10/2025, 04:27:15
class Solution {

    public void recReverse(char[] str, int s, int e){

        if(s >= e){
            return;
        }

        char temp = str[s];
        str[s] = str[e];
        str[e] = temp;

        recReverse(str, s+1, e-1);

    }

    public void reverseString(char[] s) {
        int n = s.length;
        recReverse(s, 0, n - 1);
    }
}